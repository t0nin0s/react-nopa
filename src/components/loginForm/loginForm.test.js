import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import LoginForm from './loginForm'

describe('Login Form tests', () => {
  let login;

  beforeEach(() => {
    login = mount(<LoginForm />)
  })

  test('LoginForm component renders as expected in homepage', () => {
    const component = shallow(<LoginForm home='true' />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })

  test('Surname input expected to be rendered', () => {
    const surname = login.find('#surname')
    expect(surname).toBeDefined()
  })

  test('sortcode input expected to be rendered', () => {
    const sortcode = login.find('#sortcode')
    expect(sortcode).toBeDefined()
  })

  test('account input expected to be rendered', () => {
    const account = login.find('#account')
    expect(account).toBeDefined()
  })

  test('passcode input expected to be rendered', () => {
    const passcode = login.find('#passcode')
    expect(passcode).toBeDefined()
  })

  test('memorable input expected to be rendered', () => {
    const memorable = login.find('#memorable')
    expect(memorable).toBeDefined()
  })

  test('fill out the form', () => {
    login.ref('surname').simulate('change', {target: {value: 'David'}})
    login.ref('sortcode').simulate('change', {target: {value: 'Blane'}})
    login.find('form').simulate('submit')
  })

})
