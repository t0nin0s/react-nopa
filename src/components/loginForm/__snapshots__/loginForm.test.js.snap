// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form tests LoginForm component renders as expected in homepage 1`] = `
<div>
  <div
    className="hidden"
    id="error"
  >
    There are errors, please check
  </div>
  <form
    method="post"
    onSubmit={[Function]}
  >
    <input
      className=""
      id="surname"
      name="surname"
      onChange={[Function]}
      placeholder="Surname"
      type="text"
      value=""
    />
    <input
      className=""
      id="sortcode"
      name="sortcode"
      onChange={[Function]}
      placeholder="Sort Code"
      type="text"
      value=""
    />
    <input
      className=""
      id="account"
      name="account"
      onChange={[Function]}
      placeholder="Account Number"
      type="text"
      value=""
    />
    <input
      className=""
      id="passcode"
      name="passcode"
      onChange={[Function]}
      placeholder="Passcode"
      type="text"
      value=""
    />
    <input
      className=""
      id="memorable"
      name="memorable"
      onChange={[Function]}
      placeholder="Memorable"
      type="text"
      value=""
    />
    <input
      className="button"
      type="submit"
      value="Log in & connect"
    />
  </form>
</div>
`;
